syft/rekor/queryRekor/utils.go

```diff
@@ -0,0 +1,274 @@
+		"subject:", cert.Subject.CommonName,
+		"\nsubject alternative name: email: ", cert.EmailAddresses,
+		"\nissuer: ", cert.Issuer.CommonName,
+		"\nnotBefore: ", cert.NotBefore.String(),
+		"\nnotAfter: ", cert.NotAfter.String(),
+		"\n*** certificate *** \n\n",
+	)
+	fmt.Println(s)
+}
+
+func pprintStruct(v any) {
+	json, err := json.MarshalIndent(v, "", "\t")
+	if err != nil {
+		fmt.Println(err)
+	}
+	fmt.Println(string(json))
+}
+
+// decode and parse certificate
+func parseCert(decodedCert string) (*x509.Certificate, error) {
+
+	pem, _ := pem.Decode([]byte(decodedCert))
+	if pem == nil {
+		return nil, errors.New("certificate could not be found")
+	}
+
+	cert, err := x509.ParseCertificate(pem.Bytes)
+	if err != nil {
+		log.Debug("Error parsing x509 certificate")
+		return nil, err
+	}
+	return cert, err
+}
+
+// decode and marshall log entry body
+func parseEntry(entry *models.LogEntryAnon) (*models.IntotoV001Schema, error) {
+
+	bodyEncoded, ok := entry.Body.(string)
+	if !ok {
+		return nil, errors.New("attempted to parse entry body as string, but failed")
+	}
+
+	bodyDecoded, err := base64.StdEncoding.DecodeString(bodyEncoded)
+	if err != nil {
+		log.Debug("Error base64 decoding body")
+		return nil, err
+	}
+
+	var intoto *models.Intoto = new(models.Intoto)
+	err = intoto.UnmarshalBinary(bodyDecoded)
+	if err != nil {
+		log.Debug("Error unmarshaling json entry body to intoto")
+		return nil, err
+	}
+
+	// TODO: validate
+	if *intoto.APIVersion != "0.0.1" {
+		log.Debug("Error parsing rekor entry body")
+		err := fmt.Sprintf("intoto schema version %v not supported", *intoto.APIVersion)
+		return nil, errors.New(err)
+	}
+
+	specBytes, err := json.Marshal(intoto.Spec)
+	if err != nil {
+		log.Debug("error marshaling intoto spec to json")
+		return nil, err
+	}
+
+	intotoV001 := new(models.IntotoV001Schema)
+	err = intotoV001.UnmarshalBinary(specBytes)
+	if err != nil {
+		log.Debug("Error unmarshaling intoto spec to intotoV001 schema")
+		return nil, err
+	}
+
+	return intotoV001, err
+}
+
+// parse the attestation and validate that it fits the inTotoAttestation type
+func parseAndValidateAttestations(entries []models.LogEntryAnon) ([]inTotoAttestation, error) {
+
+	var atts []inTotoAttestation
+	for _, entry := range entries {
+		attAnon := entry.Attestation
+		if attAnon == nil {
+			log.Debug("Error parsing attestation")
+			return nil, errors.New("log entry attestation is nil")
+		}
+
+		// this decodes the base64 string
+		attDecoded := string(attAnon.Data)
+
+		att := new(inTotoAttestation)
+		err := json.Unmarshal([]byte(attDecoded), att)
+		if err != nil {
+			log.Debug("Error parsing attestation")
+			return nil, err
+		}
+
+		// validation the entry is an sbom entry is done by validating attestation struct
+		if att.PredicateType == "" {
+			log.Debugf("Rekor entry %v could not be parsed as an sbom entry", *entry.LogIndex)
+			continue
+		}
+		if att.PredicateType != "http://lumjjb/sbom-documents" {
```

Comment:
create a const for the predicate type will be helpful - should be exposed as well together with the statement struct definition.